plugins/action/action.h

```diff
@@ -24,29 +38,164 @@ class Action
         UNKNOWN
     };
 
+    /**
+     * Returns a human-readable English string for an Action::Result.
+     */
     static const char *result_str(Result);
 
-    typedef std::function<void(Result)> result_callback_t;
-
+    /**
+     * Arms the drone (synchronous).
+     *
+     * Note that arming a drone normally means that the motors will spin at idle.
+     * Therefore, before arming take all safety precautions and stand clear of the drone.
+     *
+     * @return result of request
+     */
     Result arm() const;
+
+    /**
+     * Disarms the drone (synchronous).
+     *
+     * This will disarm a drone if it considers itself landed. If flying in-air, the drone should
+     * reject the disarm command. Disarming means that all motors will stop.
+     *
+     * @return result of request
+     */
     Result disarm() const;
+
+    /**
+     * Kills the drone (synchronous).
+     *
+     * This will disarm a drone no matter if still landed or still in-air.
+     * Note that the drone will fall out of the sky if you use this command while flying.
+     *
+     * @return result of request
+     */
     Result kill() const;
+
+    /**
+     * Commands the drone to take off and hover at a set takeoff altitude (synchronous).
+     *
+     * This switches the drone into position control mode and commands it to take off and hover at
+     * set takeoff altitude.
+     *
+     * @return result of request
+     */
     Result takeoff() const;
+
+    /**
+     * Commands the drone to land at the current position (synchronous).
+     *
+     * This switches the drone into position control mode and commands it to take off and hover at
+     * set takeoff altitude.
+     *
+     * @return result of request
+     */
     Result land() const;
+
+    /**
+     * Commands the drone to return to the launch (takeoff) position and land there (synchronous).
+     *
+     * This switches the drone into RTL mode which generally means it will rise up to a certain
+     * altitude to clear any obstacles before heading back to the launch (takeoff) position and
+     * land there.
+     *
+     * @return result of request
+     */
     Result return_to_launch() const;
 
+    /**
+     * Callback type for async action calls.
+     */
+    typedef std::function<void(Result)> result_callback_t;
+
+    /**
+     * Arms the drone (asynchronous).
+     *
+     * Note that arming a drone normally means that the motors will spin at idle.
+     * Therefore, before arming take all safety precautions and stand clear of the drone.
+     *
+     * @param callback function to call with result of request
+     */
     void arm_async(result_callback_t callback);
+
+    /**
+     * Disarms the drone (asynchronous).
+     *
+     * This will disarm a drone if it considers itself landed. If flying in-air, the drone should
```

Comment:
Is "in-air" a necessary distinction? (here and in the sync case?). If not, remove and just say "if flying".

Reply:
I agree.